runtime/src/comm/ofi/comm-ofi.c

```diff
@@ -2985,18 +2985,34 @@ static void init_amHandling(void);
 
 static
 void init_ofiForAms(void) {
-  //
-  // Compute the amount of space we should allow for AM landing zones.
-  // We should have enough that we needn't re-post the multi-receive
-  // buffer more often than, say, every tenth of a second.  We know from
-  // the Chapel performance/comm/low-level/many-to-one test that the
-  // comm=ugni AM handler can handle just over 150k "fast" AM requests
-  // in 0.1 sec.  Assuming an average AM request size of 256 bytes, a 40
-  // MiB buffer is enough to give us the desired 0.1 sec lifetime before
-  // it needs renewing.  We actually then split this in half and create
-  // 2 half-sized buffers (see below), so reflect that here also.
-  //
-  const size_t amLZSize = ((size_t) 40 << 20) / 2;
+
+  // Specify the amount of space we should allocate for AM landing zones. We
+  // should have enough that we needn't re-post the multi-receive buffer more
+  // often than, say, every tenth of a second. The original value was 40MB as
+  // explained in the comment below. The default is now 64MB and it's
+  // configurable, although more experiments are necessary to perform a
+  // similar size computation on more modern hardware.
+  //
+  // Original comment:
+  // We know from the Chapel performance/comm/low-level/many-to-one test that
+  // the comm=ugni AM handler can handle just over 150k "fast" AM requests in
+  // 0.1 sec.  Assuming an average AM request size of 256 bytes, a 40 MiB
+  // buffer is enough to give us the desired 0.1 sec lifetime before it needs
+  // renewing.  We actually then split this in half and create 2 half-sized
+  // buffers (see below), so reflect that here also.
+  //
+  const char *ev = "COMM_OFI_AM_LZ_SIZE";
+  size_t amLZSize = (size_t) 64 << 20;
+  const char *sizeStr = chpl_env_rt_get(ev, NULL);
+  if (sizeStr != NULL) {
+    amLZSize = chpl_env_str_to_size(ev, sizeStr, amLZSize);
+  }
```

Comment:
` chpl_env_str_to_size` returns the default when `ev` is NULL. I think this could just be:

```c
const char *ev = "COMM_OFI_AM_LZ_SIZE";
const char *sizeStr = chpl_env_rt_get(ev, NULL);
size_t  amLZSize = chpl_env_str_to_size(ev, sizeStr, (size_t) 64 << 20);
```

Reply:
Ah, actually -- we already have the wrapper for this. You can just do:

```c
const size_t amLZSize = chpl_env_rt_get_size("COMM_OFI_AM_LZ_SIZE", (size_t) 64 << 20);
```

Reply:
Good catch, I overlooked that function. I was looking at how we got the heap size and went off of that. I'll make the change.

Reply:
Yeah, I did too. Heap size is more complicated since we try to parse a `%` first and then fall back to size if it wasn't or something.